ports/atmel-samd/supervisor/port.c

```diff
@@ -496,7 +497,12 @@ void port_sleep_until_interrupt(void) {
         (void) __get_FPSCR();
     }
     #endif
-    __WFI();
+    common_hal_mcu_disable_interrupts();
+    if (!tud_task_event_ready()) {
```

Comment:
@tannewt Are there more cases here when we wouldn't want to go into WFI? Other pending tasks? This relates to trying to redo the background task stuff to add flags to indicate pending work.

But I think this is a good change for now to get things working better.

Reply:
Not that I can think of immediately. We should rethink it with the background work.